devito/propagator.py

```diff
@@ -96,6 +99,9 @@ def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, op
         # Kernel operation intensity dictionary
         self._kernel_dic_oi = {'add': 0, 'mul': 0, 'load': 0, 'store': 0, 'load_list': [], 'load_all_list': [], 'oi_high': 0, 'oi_high_weighted': 0, 'oi_low': 0, 'oi_low_weighted': 0}
 
+    # list of compilers that use GCC ivdep oly gcc 4.9 support ivdep
+    _gcc_compiler_list = ["gcc", "g++", "c++"]
```

Comment:
This should be [here](https://github.com/opesci/devito/blob/master/devito/jit_manager.py#L23). 

Purpose of Propagator: 
Loops and generating code related to loops

Purpose of JitManager:
Handle the process of compilation, linking and loading the compiled library. 

No compiler specific information needs to go inside propagator.